tests/integration/testlib/assertlib.py

```diff
@@ -25,6 +27,13 @@ def assert_state(desired_state_data):
     """Given a state, assert it against the current state."""
 
     current_state_data = netinfo.show()
+    # Ignore route and dns for assert check as the check are done in the test
+    # case code.
+    current_state_data.pop(Route.KEY, None)
+    current_state_data.pop(DNS.KEY, None)
+    desired_state_data.pop(Route.KEY, None)
```

Comment:
Since this is probably an unintended side-effect for the caller (`desired_state_data` will be changed after `assert_state()` was called I suggest to use a deepcopy here for the desired_state_data and operate on the deep copy instead. What do you think?

Reply:
Good point. Will do.

Reply:
Added:

```python
desired_state_data = copy.deepcopy(desired_state_data)
```